libs/website/forms/src/lib/subscribe-form/subscribe-form.tsx

```diff
@@ -0,0 +1,89 @@
+import { useCallback, useMemo, useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import * as yup from 'yup';
+import {
+  FormError,
+  FormFields,
+  FormState,
+  HookedFormInput,
+} from '../hooked-form-input/hooked-form-input';
+import { Button } from '@haqq/website/ui-kit';
+
+const schema = yup
+  .object({
+    email: yup
+      .string()
+      .email('Invalid email format')
+      .required('Email is required'),
+  })
+  .required();
+
+function submitForm(form: FormFields): Promise<any> {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve({ status: 200 });
+    }, 2500);
+  });
+}
+
+export function SubscribeForm() {
+  const [subscribeFormState, setSubscribeFormState] = useState<FormState>(
+    FormState.idle,
+  );
+
+  const { register, handleSubmit, formState } = useForm<FormFields>({
+    resolver: yupResolver(schema),
+  });
+
+  const onSubmit = useCallback(async (data: FormFields) => {
+    try {
+      setSubscribeFormState(FormState.pending);
+      const response = await submitForm(data);
+
+      if (response.status === 200) {
+        setSubscribeFormState(FormState.success);
+      } else {
+        setSubscribeFormState(FormState.error);
+      }
+    } catch (error) {
+      console.error(error);
+      setSubscribeFormState(FormState.error);
+    }
+  }, []);
+
+  const isFormDisabled = useMemo(() => {
+    return (
+      subscribeFormState === FormState.pending ||
+      subscribeFormState === FormState.success
+    );
+  }, [subscribeFormState]);
+
+  return (
+    <div>
+      <form
```

Comment:
add `autocomplete="off"` to prevent input autocomplete

Reply:
done